cmd2/cmd2.py

```diff
@@ -2176,120 +2239,352 @@ def _cmdloop(self) -> bool:
+            return
+
+        # Find all escaped arguments
+        escaped_matches = re.finditer(MacroArg.macro_escaped_arg_pattern, value)
+
+        while True:
+            try:
+                cur_match = escaped_matches.__next__()
+
+                # Get the number between the braces
+                cur_num = int(re.findall(MacroArg.digit_pattern, cur_match.group())[0])
+
+                arg_list.append(MacroArg(start_index=cur_match.start(), number=cur_num, is_escaped=True))
+            except StopIteration:
+                break
 
+        # Set the macro
+        result = "overwritten" if macro_name in self.macros else "created"
+        self.macros[macro_name] = Macro(name=macro_name, value=value, required_arg_count=max_arg_num, arg_list=arg_list)
+        self.poutput("Macro {!r} {}".format(macro_name, result))
+
+    def macro_delete(self, args: argparse.Namespace):
+        """ Deletes macros """
+        if args.all:
+            self.macros.clear()
+            self.poutput("All macros deleted")
+        elif not args.name:
+            self.do_help(['macro', 'delete'])
         else:
-            # Get rid of duplicates
-            arglist = utils.remove_duplicates(arglist)
+            # Get rid of duplicates and strip quotes since the argparse decorator for alias command preserves them
+            macros_to_delete = [utils.strip_quotes(cur_name) for cur_name in utils.remove_duplicates(args.name)]
 
-            for cur_arg in arglist:
-                if cur_arg in self.aliases:
-                    del self.aliases[cur_arg]
-                    self.poutput("Alias {!r} cleared".format(cur_arg))
+            for cur_name in macros_to_delete:
+                if cur_name in self.macros:
+                    del self.macros[cur_name]
+                    self.poutput("Macro {!r} deleted".format(cur_name))
                 else:
-                    self.perror("Alias {!r} does not exist".format(cur_arg), traceback_war=False)
+                    self.perror("Macro {!r} does not exist".format(cur_name), traceback_war=False)
 
-    def complete_unalias(self, text: str, line: str, begidx: int, endidx: int) -> List[str]:
-        """ Tab completion for unalias """
-        return self.basic_complete(text, line, begidx, endidx, self.aliases)
+    def macro_list(self, args: argparse.Namespace):
+        """ Lists some or all macros """
+        if args.name:
+            # Get rid of duplicates and strip quotes since the argparse decorator for alias command preserves them
+            names_to_view = [utils.strip_quotes(cur_name) for cur_name in utils.remove_duplicates(args.name)]
+
+            for cur_name in names_to_view:
+                if cur_name in self.macros:
+                    self.poutput("macro create {} {}".format(cur_name, self.macros[cur_name].value))
+                else:
+                    self.perror("Macro {!r} not found".format(cur_name), traceback_war=False)
```

Comment:
I'm not following why we're using `!r` here; applies to other places where it's just a string